src/RequestInsurance/AsyncRequests/RequestPool.php

```diff
@@ -78,6 +79,9 @@ private function convertRequestToPromise(Client $client, RequestInsurance $reque
             'headers'     => array_merge($requestInsurance->getHeadersCastToArray(), ['User-Agent' => 'RequestInsurance']),
             'body'        => $requestInsurance->payload,
             'timeout'     => $requestInsurance->getEffectiveTimeout(),
+            'on_stats'    => function (TransferStats $stats) use ($requestInsurance) {
+                $requestInsurance->setTimings($stats);
+            },
```

Comment:
```suggestion
            'on_stats'    => fn (TransferStats $stats) => $requestInsurance->setTimings($stats),
```

En kortere måde at skrive det samme på :smile:

Reply:
Mucho better